src/pages/Dashboard/Modals/SupplyWithdraw/SupplyWithdrawForm.tsx

```diff
@@ -0,0 +1,201 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState, useEffect } from 'react';
+import BigNumber from 'bignumber.js';
+import { FormikProps } from 'formik';
+import { Typography } from '@mui/material';
+import { FormValues } from 'containers/AmountForm/validationSchema';
+import { IAmountFormProps } from 'containers/AmountForm';
+import {
+  TokenTextField,
+  Delimiter,
+  InfoRow,
+  IInfoRowProps,
+  PrimaryButton,
+  ProgressBarHorizontal,
+  Tooltip,
+  ValueUpdate,
+} from 'components';
+import { useTranslation } from 'translation';
+import { Asset, TokenId } from 'types';
+import {
+  getBigNumber,
+  convertCoinsToWei,
+  convertWeiToCoins,
+  formatCentsToReadableValue,
+  format,
+  currencyFormatter,
+} from 'utilities/common';
+import { useStyles } from '../styles';
+
+interface ISupplyWithdrawFormProps extends FormikProps<FormValues> {
+  asset: Asset;
+  tokenInfo: IInfoRowProps['info'];
+  userTotalBorrowBalance: BigNumber;
+  userTotalBorrowLimit: BigNumber;
+  dailyEarnings: BigNumber;
+  onSubmit: IAmountFormProps['onSubmit'];
+  enabledButtonKey: string;
+  disabledButtonKey: string;
+  maxInput: BigNumber;
+  newBalance: (amount: BigNumber) => BigNumber;
+}
+
+export const SupplyWithdrawForm: React.FC<ISupplyWithdrawFormProps> = ({
+  values,
+  setFieldValue,
+  resetForm,
+  asset,
+  tokenInfo,
+  userTotalBorrowBalance,
+  userTotalBorrowLimit,
+  dailyEarnings,
+  maxInput,
+  enabledButtonKey,
+  disabledButtonKey,
+  newBalance,
+}) => {
+  const { id: assetId } = asset;
+  const { amount } = values;
+  const validAmount = amount && !amount.isZero() && !amount.isNaN();
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+  const [newBorrowLimit, setNewBorrowLimit] = useState<BigNumber | undefined>(undefined);
+  const [newBorrowPercent, setNewBorrowPercent] = useState(
+    userTotalBorrowBalance.div(userTotalBorrowLimit).times(100),
+  );
+  useEffect(() => resetForm, []);
```

Comment:
Because we want to clear the form when we switch tabs. The form isn't re-rendered only its contents. This is because the form value is needed in the render function which is why I broke out the contents into its own component. We could remove this by creating another component that combines the form contents and the AmountForm wrapper